org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/ClosedProjectsQueue.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.builder.impl;
+
+import java.util.Deque;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import org.eclipse.emf.common.util.URI;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Singleton;
+
+/**
+ * A queue of resources that are supposed to be removed from the Xtext index.
+ * The work is encapsulated by a ClosedProjectsQueue.Task.
+ * 
+ * @since 2.17
+ */
+@Singleton
+public class ClosedProjectsQueue {
+
+	/**
+	 * Encapsulates the changes that need to be performed after one or more projects have been removed / closed.
+	 */
+	public class Task {
+		/**
+		 * The names of the projects as encapsulated by this task.
+		 */
+		public final ImmutableSet<String> projectNames;
```

Comment:
public final fields instead of getters intentional?

Reply:
Yes, that was intentional. If you prefer getters, I can do that.